aten/src/ATen/core/TensorBase.h

```diff
@@ -43,7 +43,6 @@ inline bool variable_excluded_from_dispatch() {
   // Please read the comment in `VariableFallbackKernel.cpp` about the background of this change.
   return true;
 #else
-  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(!c10::impl::tls_local_dispatch_key_set().excluded_.has(DispatchKey::Autograd));
```

Comment:
I'm not actually sure that this is asserting the right thing. `DispatchKey::Autograd` is an alias key, and it (shouldn't be) possible to represent an alias key in a DispatchKeySet. A keyset only has 64 bits, and each of those bits is meant to be used for either a backend bit or functionality bit - there aren't meant to be any bits that are reserved for alias keys. This lets us have up to 64 total backend + functionality bits, and more alias keys on top of that.

Before this PR, I'm... pretty sure that adding an alias key to a DispatchKeySet was undefined behavior? (current logic in master is [here](https://github.com/pytorch/pytorch/blob/13ea2cb330539ea386d38bf1359a537087be6f8a/c10/core/DispatchKeySet.h#L56)). We would blindly cast the DispatchKey to a `uint8_t`, which would overflow if the key was an alias key with enum value > 64. So... I'm not actually convinced that this assert would catch anything useful.

In this PR, the `DispatchKeySet` contructor will end up setting `repr_` to `0` if the `DispatchKey` argument is an alias key. That still doesn't provide any error checking (since I couldn't put error checking directly in the constructor, since it's constexpr), but at least it's not UB. But if you have any other thoughts on that let me know!

Reply:
From offline discussion, this most likely never did the right thing. So we should do some git blame investigation to try and get a clearer idea of what the intent was.

Reply:
It looks like it was added here: https://github.com/pytorch/pytorch/pull/44268.

It looks like that PR fixed a bug where the alias key was accidentally getting added to the exclude set instead of the actual runtime keys. I bet at that point, we had enough dispatch key space remaining that `DispatchKey::Autograd` was < 64, so it could get away with taking a valid bit in `DispatchKeySet`, which would let us check for that type of error. That error would have stopped being useful though once we added enough dispatch keys for `DispatchKey::Autograd` to become >= 64.

Reply:
Sounds good!